src/hotspot/share/cds/filemap.cpp

```diff
@@ -1549,11 +1550,22 @@ size_t FileMapRegion::used_aligned() const {
   return align_up(used(), MetaspaceShared::core_region_alignment());
 }
 
-void FileMapRegion::init(int region_index, size_t mapping_offset, size_t size, bool read_only,
+void FileMapRegion::init(int region_index, size_t mapping_offset, char* dumptime_base, size_t size, bool read_only,
                          bool allow_exec, int crc) {
   _is_heap_region = HeapShared::is_heap_region(region_index);
   _is_bitmap_region = (region_index == MetaspaceShared::bm);
+
+  if (_is_heap_region) {
```

Comment:
Explain please.  Only heap region has a non-null dumptime base?

Reply:
ie add a comment here too.

Reply:
I added comments here. There are also comments in cds.h that explains what these fields should be for different types of regions.